### PR TITLE
PyTorch and NiftyNet integration

### PR DESCRIPTION
**Pull request by OeslleLucena**
## Status
**WORK IN PROGRESS**

## Description
Using NiftyNet's IO modules with PyTorch

## Types of changes

- [ ] New feature (non-breaking change which adds functionality)
- [ ] My code follows the code style of this project.
- [ ] I have added tests to cover my changes.
- [ ] All new and existing tests passed.

## Todos
- [ ] Tests
- [x] Documentation

## Impacted Areas in Application
a new demo

### CODE REVIEW

**Comment by OeslleLucena** [comment]
Nice example using PyTorch and NiftyNet for 3D segmentation.

File: demos/PyTrochNiftyNet/segmentation.py [hunk 1]
```diff
@@ -0,0 +1,261 @@
...
+
+                inputs = inputs.to(device)
+                labels = labels.to(device)
+
+                # zero the parameter gradients
+                optimizer.zero_grad()
+
+                # forward
+                # track history if only in train
+                with torch.set_grad_enabled(phase == 'train'):
```
Comment (tomvars) [inline demos/PyTrochNiftyNet/segmentation.py]: Change `train` to `training`

File: demos/PyTrochNiftyNet/segmentation.py [hunk 2]
```diff
@@ -0,0 +1,261 @@
...
+            epoch_acc = running_corrects / (epoch_samples)
+
+            print('{} Loss: {:.4f} Dice: {:.4f}'.format(phase, epoch_loss, epoch_acc))
+
+            if epoch == 0:
+                best_loss = epoch_loss
+                torch.save(model.state_dict(), cp_path.format(epoch + 1))
+
+            # deep copy the model
+            if phase == 'val' and epoch_loss < best_loss:
```
Comment (tomvars) [inline demos/PyTrochNiftyNet/segmentation.py]: Change `val` to `validation`

**Review by tomvars — COMMENTED** [review]
Great stuff! Would be great if the imports could be wrapped in a `try` `except` block to avoid having torch as a dependency but have the code in the main repo.

**Pull request merged**